sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer.py

```diff
@@ -0,0 +1,163 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import unicode_literals
+
+import uuid
+import logging
+
+import uamqp  # type: ignore
+from uamqp import types, errors, utils  # type: ignore
+from uamqp import ReceiveClient, Source  # type: ignore
+
+from .exceptions import _error_handler
+from ._common import EventData, EventPosition
+from ._client_base import ConsumerProducerMixin
+from ._utils import create_properties, trace_link_message
+from ._constants import (
+    EPOCH_SYMBOL,
+    TIMEOUT_SYMBOL,
+    RECEIVER_RUNTIME_METRIC_SYMBOL,
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EventHubConsumer(ConsumerProducerMixin):  # pylint:disable=too-many-instance-attributes
+    """
+    A consumer responsible for reading EventData from a specific Event Hub
+    partition and as a member of a specific consumer group.
+
+    A consumer may be exclusive, which asserts ownership over the partition for the consumer
+    group to ensure that only one consumer from that group is reading the from the partition.
+    These exclusive consumers are sometimes referred to as "Epoch Consumers."
```

Comment:
Doc nit:  When I had to go through and set up the epoch consumer, I didn't know that _really_ meant to look for "owner_level."

We might want to add a note here, something like "Create an epoch consumer by defining an owner_level to the optional parameter" (It does get mentioned before that it's an "exclusive consumer" but we mix our naming a bit in this space)

Reply:
This parts of the docs isn't actually exposed - but agreed we need better docs on this.
Will do a separate docs pass once this refactor is out of the way.